File: backend/src/controllers/inquiry.controller.ts
import { Request, Response } from 'express';
import { InquiryService } from '../services/inquiry.service';

interface AuthRequest extends Request {
  userId?: number;
}

export class InquiryController {
  private inquiryService = new InquiryService();

  getAll = async (req: Request, res: Response) => {
    try {
      const page = parseInt(req.query.page as string) || 1;
      const limit = parseInt(req.query.limit as string) || 10;
      const result = await this.inquiryService.getAll(page, limit);
      res.json(result);
    } catch (error: any) {
      res.status(500).json({ message: '문의글 목록 조회 실패' });
    }
  };

  getById = async (req: Request, res: Response) => {
    try {
      const id = parseInt(req.params.id);
      const inquiry = await this.inquiryService.getById(id);
      if (!inquiry) {
        return res.status(404).json({ message: '문의글을 찾을 수 없습니다' });
      }
      res.json(inquiry);
    } catch (error: any) {
      res.status(500).json({ message: '문의글 조회 실패' });
    }
  };

  create = async (req: AuthRequest, res: Response) => {
    try {
      const { title, content } = req.body;
      const inquiry = await this.inquiryService.create(title, content, req.userId!);
      res.status(201).json(inquiry);
    } catch (error: any) {
      res.status(400).json({ message: error.message });
    }
  };

  update = async (req: AuthRequest, res: Response) => {
    try {
      const id = parseInt(req.params.id);
      const { title, content } = req.body;
      const inquiry = await this.inquiryService.update(id, title, content, req.userId!);
      res.json(inquiry);
    } catch (error: any) {
      if (error.message === '권한이 없습니다') {
        return res.status(403).json({ message: error.message });
      }
      res.status(400).json({ message: error.message });
    }
  };

  delete = async (req: AuthRequest, res: Response) => {
    try {
      const id = parseInt(req.params.id);
      await this.inquiryService.delete(id, req.userId!);
      res.json({ message: '문의글이 삭제되었습니다' });
    } catch (error: any) {
      if (error.message === '권한이 없습니다') {
        return res.status(403).json({ message: error.message });
      }
      res.status(400).json({ message: error.message });
    }
  };
}

File: backend-java/src/main/java/com/inquiry/controller/InquiryController.java
package com.inquiry.controller;

import com.inquiry.dto.request.InquiryRequest;
import com.inquiry.dto.response.InquiryResponse;
import com.inquiry.dto.response.PageResponse;
import com.inquiry.service.InquiryService;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.security.core.Authentication;
import org.springframework.web.bind.annotation.*;

import java.util.Map;

@RestController
@RequestMapping("/api/inquiries")
@RequiredArgsConstructor
public class InquiryController {

    private final InquiryService inquiryService;

    @GetMapping
    public ResponseEntity<PageResponse<InquiryResponse>> getAll(
            @RequestParam(defaultValue = "1") int page,
            @RequestParam(defaultValue = "10") int limit) {
        return ResponseEntity.ok(inquiryService.getAll(page, limit));
    }

    @GetMapping("/{id}")
    public ResponseEntity<InquiryResponse> getById(@PathVariable Long id) {
        return ResponseEntity.ok(inquiryService.getById(id));
    }

    @PostMapping
    public ResponseEntity<InquiryResponse> create(
            @Valid @RequestBody InquiryRequest request,
            Authentication authentication) {
        Long userId = (Long) authentication.getPrincipal();
        InquiryResponse response = inquiryService.create(request, userId);
        return ResponseEntity.status(HttpStatus.CREATED).body(response);
    }

    @PutMapping("/{id}")
    public ResponseEntity<InquiryResponse> update(
            @PathVariable Long id,
            @Valid @RequestBody InquiryRequest request,
            Authentication authentication) {
        Long userId = (Long) authentication.getPrincipal();
        return ResponseEntity.ok(inquiryService.update(id, request, userId));
    }

    @DeleteMapping("/{id}")
    public ResponseEntity<Map<String, String>> delete(
            @PathVariable Long id,
            Authentication authentication) {
        Long userId = (Long) authentication.getPrincipal();
        inquiryService.delete(id, userId);
        return ResponseEntity.ok(Map.of("message", "문의글이 삭제되었습니다"));
    }
}
